codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/KotlinWriter.kt

```diff
@@ -219,7 +220,9 @@ class KotlinWriter(
      */
     private fun renderDeprecatedAnnotation(shape: Shape) {
         if (shape.isDeprecated) {
-            write("""@Deprecated("No longer recommended for use. See AWS API documentation for more details.")""")
+            val message = shape.expectTrait<DeprecatedTrait>().message?.getOrNull()
+                ?: "No longer recommended for use. See AWS API documentation for more details."
+            write("""@Deprecated("$message")""")
```

Comment:
**Correctness**: Unsafe string interpolation. If `message` contains double-quotes this will generate syntax errors. Use `#S` instead:

```kotlin
write("""@Deprecated("#S")""", message)
```

Reply:
Good catch, thanks!